src/net.c

```diff
@@ -494,6 +494,10 @@ coap_option_check_critical(coap_context_t *ctx,
 void
 coap_transaction_id(const coap_address_t *peer, const coap_pdu_t *pdu, 
 		    coap_tid_t *id) {
+#if !defined(WITH_POSIX) && !defined(WITH_LWIP) && !defined(WITH_CONTIKI)
+  (void)peer;
```

Comment:
In that case we could make the whole *coap_transaction_id*() unavailable if neither of these *WITH_** macros is defined. Otherwise *peer* is not used anyway and the result doesn't change.